clang/test/CodeGenSYCL/device_global_int_footer_header.cpp

```diff
@@ -55,6 +55,9 @@ int main() {
 device_global<int> Basic;
 // CHECK-FOOTER-NEXT: device_global_map::add((void *)&::Basic, "_Z5Basic");
 
+extern device_global<int> ExternDevGlobal;
+// CHECK-FOOTER-NOT: ExternDevGlobal
+
```

Comment:
Could you add a variant where it is declared extern, followed by a regular definition?